lib/Runtime/Types/RecyclableObject.h

```diff
@@ -313,6 +313,7 @@ namespace Js {
         virtual BOOL InitPropertyInEval(PropertyId propertyId, Var value, PropertyOperationFlags flags = PropertyOperation_None, PropertyValueInfo* info = NULL);
         virtual BOOL EnsureProperty(PropertyId propertyId);
         virtual BOOL EnsureNoRedeclProperty(PropertyId propertyId);
+        virtual BOOL EnsureNoRedeclFunc(PropertyId propertyId);
```

Comment:
I see that you're following the pattern of EnsureNoRedeclProperty, but I don't think the pattern should necessarily apply, because EnsureNoRedeclProperty is used in a much bigger variety of ways.

* Why does it have a return value? As far as I can tell, the returned value is never used.
* Why is this virtual and defined on RecyclableObject? The only call site knows that the type will be RootObjectBase; can't we just cast directly to that and call a non-virtual method?